### PR TITLE
Feature/skip auth provider selection

### PR DESCRIPTION
**Pull request by machisuji**
OP [WP #7629](https://www.openproject.org/work_packages/7629)

### CODE REVIEW

**Comment by machisuji** [comment]
mp and org config green too

File: app/controllers/concerns/omniauth_login.rb [hunk 1]
```diff
@@ -24,6 +24,28 @@ def omniauth_failure
     redirect_to :action => 'login'
   end
 
+  def self.direct_login?
+    direct_login_provider.is_a? String
```
Comment (linki) [inline app/controllers/concerns/omniauth_login.rb:28]: isn't a provider set when this value is not nil? could we just test against `direct_login_provider` or `!!direct_login_provider` here?

File: config/configuration.yml.example [hunk 2]
```diff
@@ -170,6 +170,22 @@ default:
...
+  # With his configuration option you can set a specific omniauth provider to be
+  # used for direct login. Meaning that the login provider selection is skipped and
+  # the configured provider is used directly instead.
+  #
+  # If this option is active /login will lead directly to the configured omniauth provider
+  # and so will a click on 'Sign in' (as opposed to opening the drop down menu).
+  #
+  # Note that this does not stop a user from manually navigating to any other
+  # omniauth provider if additional ones are configured.
+  # omniauth_direct_login_provider: developer
```
Comment (linki) [inline config/configuration.yml.example:188]: how about a symbol here? `:developer`

File: doc/CONFIGURATION.md [hunk 3]
```diff
@@ -62,12 +62,28 @@ In case you want to use environment variables, but you have no easy way to set t
...
+* [`omniauth_direct_login_provider`](#omniauth-direct-login-provider) (default: nil)
 
-Email configuration
+### omniauth direct login provider
+
+*default: nil*
+
+Example:
+
+    omniauth_direct_login_provider: google
```
Comment (linki) [inline doc/CONFIGURATION.md:73]: and here?

File: spec/controllers/account_controller_spec.rb [hunk 4]
```diff
@@ -156,6 +156,27 @@
...
+    end
+
+    describe 'POST' do
+      it 'redirects to some_provider' do
+        post :login, username: 'foo', password: 'bar'
+
+        expect(response).to redirect_to '/auth/some_provider'
+      end
+    end
+  end
```
Comment (linki) [inline spec/controllers/account_controller_spec.rb:179]: :+1:

File: spec/views/layouts/base.html.erb_spec.rb [hunk 5]
```diff
@@ -37,4 +37,33 @@
...
+        expect(OmniauthLogin).to receive(:direct_login_provider).and_return('some_provider')
+        render
+      end
+
+      it 'shows just a sign-in link, no menu' do
+        expect(response).to have_selector "a[href='/login']"
+        expect(response).not_to have_selector "div[id='nav-login-content']"
+      end
+    end
+  end
```
Comment (linki) [inline spec/views/layouts/base.html.erb_spec.rb:68]: :+1:

**Comment by linki** [comment]
just some optional remarks. I like the tests :+1:

**Comment by linki** [comment]
:kissing_closed_eyes:

**Pull request merged**